### PR TITLE
Update fluxc to SiteStore using coroutines

### PR DESCRIPTION
**Pull request by malinajirka**
This PR updates fluxc to https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2018.

To test:
Smoke test the app including login - @planarvoid if you have tips on which parts we should test, please update the description

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by planarvoid** [comment]
Thanks for the PR 👍 , I'd test mainly login and loading site list and account. I've created a Woo store but I'm not really familiar with the app enough to test it. From my POV it's looking good @malinajirka

**Comment by malinajirka** [comment]
I've rested several flows and all LGTM!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/push/NotificationHandler.kt [hunk 1]
```diff
@@ -346,26 +346,28 @@ class NotificationHandler @Inject constructor(
 
     @VisibleForTesting
     fun dispatchNewOrderEvents(model: NotificationModel) {
-        dispatcher.dispatch(
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/push/NotificationHandler.kt:349]: @planarvoid I had to update this code as `siteStore.getSiteBySiteId` returns nullable SiteModel now (tbh I'm quite surprised it was retuning non-nullable model before).  When this code is reviewed, I think we are ready to merge the PR.

Comment (malinajirka) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/push/NotificationHandler.kt:349]: @wzieba Could you please double check this change. You mentioned [here](https://github.com/woocommerce/woocommerce-android/pull/4204#discussion_r656992773) that we need to fetch the order twice. But AFAIU it's ok to load the SiteModel just once, right?

Comment (wzieba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/push/NotificationHandler.kt:349]: Thanks @malinajirka for ping! Yes you're right, it's ok to load SiteModel just once.

Comment (malinajirka) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/push/NotificationHandler.kt:349]: Thanks! ;)

**Review by planarvoid — APPROVED** [review]
as far as I can tell the changes are looking good 👍

**Pull request merged**